File: app/src/main/java/com/hk/ijournal/repository/local/UserDao.java
package com.hk.ijournal.repository.local;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.hk.ijournal.repository.models.DiaryUser;

@Dao
public interface UserDao {
    @Insert
    void insertUser(DiaryUser user);

    @Delete
    void deleteUser(DiaryUser user);

    @Query("select * from usertable where username = :userName")
    DiaryUser getUserbyName(String userName);
}

File: app/src/main/java/com/hk/ijournal/repository/local/UserDao.kt
package com.hk.ijournal.repository.local

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.hk.ijournal.repository.models.DiaryUser

@Dao
interface UserDao {
    @Insert
    fun insertUser(user: DiaryUser)

    @Delete
    fun deleteUser(user: DiaryUser)

    @Query("select * from usertable where username = :userName")
    fun getUserbyName(userName: String?): DiaryUser?
}